packages/drivers/odsp-driver/src/test/getVersions.spec.ts

```diff
@@ -0,0 +1,178 @@
+            localCache,
+            {
+                docId: hashedDocumentId,
+                resolvedUrl,
+            },
+            new TelemetryNullLogger(),
+        );
+
+        const resolved = await resolver.resolve({ url: odspUrl });
+        service = new OdspDocumentStorageService(
+            resolved,
+            async (_options) => "token",
+            logger,
+            true,
+            { ...nonPersistentCache, persistedCache: epochTracker},
+            hostPolicy,
+            epochTracker,
+            async () => { return {}; },
+            );
+    });
+
+    it("cache fetch throws and network fetch succeeds", async () => {
+        // overwriting get() to make cache fetch throw
+        localCache.get = async () => {
+            throw new Error("testing");
+        };
+
+        const version = await mockFetchSingle(
+            async () => service.getVersions(null,1),
+            async () => createResponse({ "x-fluid-epoch": "epoch1" }, odspSnapshot, 200),
+        );
+
+        assert.deepStrictEqual(version, [{ id: "id", treeId: undefined!}], "incorrect version");
+    });
+
+    it("cache fetch succeeds and network fetch succeeds", async () => {
+        const cacheEntry: ICacheEntry = {
+            key:"",
+            type: "snapshot",
+            file: { docId: hashedDocumentId, resolvedUrl } };
+        await localCache.put(cacheEntry, value);
+
+        const version = await mockFetchSingle(
+            async () => service.getVersions(null,1),
+            async () => createResponse({ "x-fluid-epoch": "epoch1" }, odspSnapshot, 200),
+        );
+        assert.deepStrictEqual(version, [], "incorrect version");
+    });
+
+    it("cache fetch throws and network fetch throws", async () => {
+        // overwriting get() to make cache fetch throw
+        localCache.get = async () => {
+            throw new Error("testing");
+        };
+
+        let isCaught = false;
+        try { await mockFetchSingle(
+            async () => service.getVersions(null,1),
+            // 404 response expected so network fetch throws
+            notFound,
+        );
+        } catch (error) {
+            isCaught = true;
+            assert.strictEqual(error.message, "Error 404 (undefined)", "incorrect error message");
+        }
+        // making sure network fetch did throw and catch block was executed
+        assert(isCaught, "catch block was not executed");
+    });
+
+    it("cache fetch succeeds and network fetch throws", async () => {
+        const cacheEntry: ICacheEntry = {
+            key:"",
+            type: "snapshot",
+            file: { docId: hashedDocumentId, resolvedUrl } };
+        await localCache.put(cacheEntry, value);
+
+        const version = await mockFetchSingle(
+            async () => service.getVersions(null,1),
+            // 404 response expected so network fetch throws
+            notFound,
+        );
+        assert.deepStrictEqual(version, [], "incorrect version");
```

Comment:
How come if cache fetch succeeds and still you are getting empty version?

Reply:
because the value i put in the cache is not a odspSnapshotCacheValue, so there is no id, and it returns empty

Reply:
Can you put the appropriate odspSnapshotCacheValue, so that it returns something instead of []?